skimage/metrics/tests/test_structural_similarity.py

```diff
@@ -257,3 +261,26 @@ def test_invalid_input():
         structural_similarity(X, X, K2=-0.1)
     with pytest.raises(ValueError):
         structural_similarity(X, X, sigma=-1.0)
+
+
+def test_ms_structural_similarity():
+    # color image example
+    Xc = data.chelsea()
+    sigma = 15.0
+    Yc = np.clip(Xc + sigma * np.random.randn(*Xc.shape), 0, 255)
+    Yc = Yc.astype(Xc.dtype)
+
+    # multichannel result should be mean of the individual channel results
+    mssim = multiscale_structural_similarity(Xc, Yc, channel_axis=-1)
+    mssim_sep = [multiscale_structural_similarity(
+        Yc[..., c], Xc[..., c]) for c in range(Xc.shape[-1])]
+    assert_almost_equal(mssim, np.mean(mssim_sep))
+
+    # structural_similarity of image with itself should be 1.0
+    assert_equal(multiscale_structural_similarity(Xc, Xc, channel_axis=-1), 1.0)
+
+    # test small image
+
+    X = np.ones((5, 5))
+    with pytest.raises(AssertionError):
+        multiscale_structural_similarity(X, X)
```

Comment:
I didn't get this test... shouldn't it be `1` here? As in, isn't the multiscale structural similarity of an image w/ itself equal to `1`?

Reply:
You are right the MS-SSIM of an image with itself is 1.

By default in the function 4 scales are used so the minimum image size required is 2**4. Here the image is smaller only 5, so it will throw an  `AssertionError`.
This is the error I am trying to evoke-
https://github.com/a-parida12/scikit-image/blob/main/skimage/metrics/_structural_similarity.py#L309